Update the `common` gadgets of `circuit`

duguorong009 opened the pull request:
What's been done:
 - Update the `common` gadgets so that they make use of new `CommonConfig`, `MainConfig` & `MainChip`
 - Update the related `circuit` gadgets so that they make use of new `common` gadgets

lazovicff commented on circuit/src/circuit.rs:
```diff
@@ -207,8 +206,9 @@ impl<
-		let lt_eq = LessEqualConfig::new(bits2num_selector, n_shifted_selector, is_zero_selector);
+		let main_selector = MainChip::configure(&common, meta);
+		let main = MainConfig::new(main_selector);
+		let lt_eq = LessEqualConfig::new(bits2num_selector, n_shifted_selector, main.clone());
```
Can we follow the convention and have chipset configs first, then selectors after. When defining constructors?
Apply in all cases

duguorong009 replied:
Done in https://github.com/eigen-trust/protocol/pull/145/commits/948ef3472265ab9d1fd0cbbfcc08cd35154c0c1c

lazovicff replied:
I meant in the constructors, like so:
```rust
LessEqualConfig::new(main.clone(), bits2num_selector, n_shifted_selector)
```

duguorong009 replied:
Thanks for the indication! @lazovicff 
Done in https://github.com/eigen-trust/protocol/pull/145/commits/b1e552983ab6faf31f59f2dbdb557ac0513e263e

lazovicff commented:
2 more things:
- Can we rename `common.rs` to `main.rs`
- Can we replace this: https://github.com/eigen-trust/protocol/blob/update-circuit-common/circuit/src/lib.rs#L175-L190 with this:
```rust
impl CommonConfig {
	/// Create a new `CommonConfig` columns
	pub fn new<F: FieldExt>(meta: &mut ConstraintSystem<F>) -> Self {
		let advice = [(); ADVICE].map(|_| meta.advice_column());
		let fixed = [(); FIXED].map(|_| meta.fixed_column());
		let instance = meta.instance_column();

		advice.map(|c| meta.enable_equality(c));
		fixed.map(|c| meta.enable_constant(c));
		meta.enable_equality(instance);

		Self { advice, fixed, instance }
	}
}
```

duguorong009 commented:
> 2 more things:
> 
> * Can we rename `common.rs` to `main.rs`
> * Can we replace this: https://github.com/eigen-trust/protocol/blob/update-circuit-common/circuit/src/lib.rs#L175-L190 with this:
> 
> ```rust
> impl CommonConfig {
> 	/// Create a new `CommonConfig` columns
> 	pub fn new<F: FieldExt>(meta: &mut ConstraintSystem<F>) -> Self {
> 		let advice = [(); ADVICE].map(|_| meta.advice_column());
> 		let fixed = [(); FIXED].map(|_| meta.fixed_column());
> 		let instance = meta.instance_column();
> 
> 		advice.map(|c| meta.enable_equality(c));
> 		fixed.map(|c| meta.enable_constant(c));
> 		meta.enable_equality(instance);
> 
> 		Self { advice, fixed, instance }
> 	}
> }
> ```

Done in https://github.com/eigen-trust/protocol/pull/145/commits/948ef3472265ab9d1fd0cbbfcc08cd35154c0c1c
